### PR TITLE
Fix deprecated Grafana dashboards

### LINKED ISSUE #2130: [1] Upgrade Grafana Dashboards with deprecated Angular plugins
### Chart to upgrade

grafana-dashboards

### Reason for upgrade

- [ ] Security patching
- [ ] Upgrade to allow new features
- [X] Upgrade to prevent future issues
- [X] Upgrade to maintain a supported version
- [ ] Other - _Write the reason below_

### Additional context

[Grafana is dropping support for Angular plugins](https://grafana.com/docs/grafana/latest/developers/angular_deprecation/). A bunch of our dashboards are affected by this, and with the more recent versions of our apps this can be seen when checking a dashboard as they contain these warnings for affected panels: 
image

We should go through and update our dashboards to avoid future issues. Preferably, we want this done before we upgrade to grafana v12. 

> [!TIP]
> For most affected panels it is possible to migrate them when editing them via the UI: 
> image

Doing a quick check in a `v0.37` environment, these dashboards seems to be affected (might have missed some):
- [ ] Alertmanager / Overview
- [x] Backup status
- [ ] Cluster Autoscaler Stats
- [x] Daily checks
- [ ] Fluentd 1.x
- [ ] Grafana Overview
- [ ] Harbor
- [ ] kubernetes-mixin dashboards
- [ ] NGINX Ingress Controller
- [ ] node-exporter-mixin dashboards
- [ ] Node Local DNS
- [ ] prometheus-mixin dashboards
- [ ] Prometheus / Time series
- [ ] Prometheus Blackbox Exporter
- [ ] thanos-mixin dashboards
- [x] Velero Stats

### Acceptance criteria

- I checked the migration of the new chart version:
  - [ ] The upgrade does not require migration steps
  - [ ] The upgrade requires migration steps
- I tested the functionality of the new chart version:
  - [ ] The upgrade does not change current functionality
  - [ ] The upgrade changes current functionality

### PR DESCRIPTION
**Pull request by anders-elastisys**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

This PR updates Grafana Dashboards containing deprecated panels.

`node-mixin` was uncommented from the [grafana-dashboards makefile](https://github.com/elastisys/compliantkubernetes-apps/pull/2242/commits/d2638d471c528205e323258d824c61a72cee7fc2#diff-a0a228277fbe90de8e2699cb323c863fd44bba9f43dfdd1c9a18b55c4d36d5ac) and [adjusted](https://github.com/elastisys/compliantkubernetes-apps/pull/2242/commits/c5e98450bfa58973ee174d7e16cc810f5796dc1c#diff-e4598d2f497d0cc25cdd88039654da56503cd8015a914deab2cd024f79fde022) and seems to be working fine now, but the latest upstream dashboards still included deprecated panels and were fixed manually. There exists an issue regarding this: https://github.com/prometheus/node_exporter/issues/3046. Also, running the `make` script for `node-mixin` pulled down two new dashboards, one which was exclusive for darwin/macOS which I removed manually as I am not sure how to configure `jsonnet` to only include certain files. The other dashboard was `Node Exporter / USE Method / Multi-cluster` which I kept in.

`alertmanager-mixin` did not get upgraded with make script, and can't see issues fixing the panels in the repo https://github.com/grafana/prometheus-alertmanager.

`prometheus-mixin` did not get upgraded with make script. There exists an issue regarding fixing deprecated panels: https://github.com/prometheus/prometheus/issues/14404

`thanos-mixin` did not get upgraded with make script. Issue regarding fixing deprecated panels: https://github.com/thanos-io/thanos/issues/7413

Also noticed that the fluentd dashboard was missing metrics used in some panels, see [this commit](https://github.com/elastisys/compliantkubernetes-apps/pull/2242/commits/1190295bd8d13aa2c02869f60215425b5f6b87cf) for newly added config to get this metric. 

- Fixes #2130

#### Information to reviewers

Not sure if the migration script will be necessary, planning to upgrade Grafana to v11 once this gets merged, which will probably make the script obsolete.

I manually checked individual panels when migrating them that they looked close to the original, some panels needed more adjustments than others. Noticed we have some panels with metrics (besides the fluentd one I added) that I did at least not have in my dev environment, might be worth looking into. Might have missed something as there were a lot of dashboards and panels to go through. :sweat_smile: 

Tested running [detect-angular-dashboards](https://github.com/grafana/detect-angular-dashboards) after having migrated and it did not detect any deprecated panels.

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [x] The change is disruptive
  - [ ] The change requires no migration steps
  - [x] The change requires migration steps
  - [ ] The change upgrades CRDs
  - [ ] The change updates the config *and* the schema
- Metrics checks:
  - [x] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Review by robinelastisys — APPROVED** [review]
Looks like its going in the right direction! Will be awesome to get [this](https://github.com/elastisys/compliantkubernetes-apps/issues/2130) fixed! 

Only thing im slightly worried about is this:

> node-mixin was uncommented from the [grafana-dashboards makefile](https://github.com/elastisys/compliantkubernetes-apps/pull/2242/commits/d2638d471c528205e323258d824c61a72cee7fc2#diff-a0a228277fbe90de8e2699cb323c863fd44bba9f43dfdd1c9a18b55c4d36d5ac) and [adjusted](https://github.com/elastisys/compliantkubernetes-apps/pull/2242/commits/c5e98450bfa58973ee174d7e16cc810f5796dc1c#diff-e4598d2f497d0cc25cdd88039654da56503cd8015a914deab2cd024f79fde022) and seems to be working fine now, but the latest upstream dashboards still included deprecated panels and were fixed manually

Not sure if thats the way to go but i let others have opinions on that!

File: helmfile.d/values/fluentd/forwarder-common.yaml.gotmpl [hunk 1]
```diff
@@ -233,6 +233,21 @@ extraConfigMaps:
       @type kubernetes_metadata
     </filter>
 
+    # count the number of incoming records per tag
```
Comment (OlleLarsson) [inline helmfile.d/values/fluentd/forwarder-common.yaml.gotmpl:236]: This seems a bit out-of-scope for this PR. Could it be extracted to its own PR?
Is this a nicely behaved metric or do we risk [cardinality explosion](https://grafana.com/blog/2022/02/15/what-are-cardinality-spikes-and-why-do-they-matter/)?

Comment (anders-elastisys) [reply helmfile.d/values/fluentd/forwarder-common.yaml.gotmpl:236]: I can move it into its own PR and it can be discussed there, I have not looked into the cardinality of the metric. But if we don't want the metric, if it misbehaves or is redundant than we should remove those fluentd panels instead, but that can be handled in a separate PR.

**Review by Zash — APPROVED** [review]

**Review by lunkan93 — APPROVED** [review]
